src/lib/message/L1ToL2Message.ts

```diff
@@ -688,6 +701,73 @@ export class EthDepositMessage {
     )
   }
 
+  /**
+   * Parse the data field in
+   * event InboxMessageDelivered(uint256 indexed messageNum, bytes data);
+   * @param eventData
+   * @returns
+   */
+  private static parseEthDepositData(eventData: string): BigNumber {
+    // https://github.com/OffchainLabs/nitro/blob/9f16d082496aef9de66b9e8653531d467d685560/contracts/src/bridge/Inbox.sol#L230
+    const parsed = ethers.utils.defaultAbiCoder.decode(
+      ['uint256'],
+      eventData
+    ) as [BigNumber]
+
+    return parsed[0]
+  }
+
+  /**
+   * Create an eth deposit message from a transaction receipt that called ethDeposit
+   * in Inbox.sol
+   * @param l2Provider
+   * @param chainId
+   * @param txReceipt Receipt for the transaction originating the events
+   * @param inboxMessageEvent The InboxMessageDeliveredEvent emitted from the depositing transaction
+   * @param bridgeMessageEvent The MessageDeliveredEvent emitted from Bridge.sol in the depositing transaction
+   * @returns
+   */
+  public static fromTxReceipt(
+    l2Provider: Provider,
+    chainId: number,
```

Comment:
we can simplify the api a bit by making this function async then inferring the chainId from the provider
```ts
    const chainID = (await l2Provider.getNetwork()).chainId
```